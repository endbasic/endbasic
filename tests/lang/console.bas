' EndBASIC
' Copyright 2020 Julio Merino
'
' Licensed under the Apache License, Version 2.0 (the "License"); you may not
' use this file except in compliance with the License.  You may obtain a copy
' of the License at:
'
'     http://www.apache.org/licenses/LICENSE-2.0
'
' Unless required by applicable law or agreed to in writing, software
' distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
' WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
' License for the specific language governing permissions and limitations
' under the License.

' Tests screen manipulation commands.
'
' Note that the output of this test stored in `console.out` contains control
' characters and that we simply expect that they produce the correct visual
' effect.  For that reason, whenever modifying this test, make sure to run it
' by hand and observe if it's doing the right thing before overwriting the
' golden data file.

PRINT "Before clearing the screen"
CLS
PRINT "After clearing the screen"

row = 0
WHILE row < 10
    column = 0
    WHILE column < 20
        LOCATE row + 5, column + 10
        PRINT "#"
        column = column + 1
    END WHILE
    row = row + 1
END WHILE

LOCATE 100, 0
