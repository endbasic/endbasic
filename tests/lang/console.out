Before clearing the screen
[2J[1;1HAfter clearing the screen
[6;11H#
[6;12H#
[6;13H#
[6;14H#
[6;15H#
[6;16H#
[6;17H#
[6;18H#
[6;19H#
[6;20H#
[6;21H#
[6;22H#
[6;23H#
[6;24H#
[6;25H#
[6;26H#
[6;27H#
[6;28H#
[6;29H#
[6;30H#
[7;11H#
[7;12H#
[7;13H#
[7;14H#
[7;15H#
[7;16H#
[7;17H#
[7;18H#
[7;19H#
[7;20H#
[7;21H#
[7;22H#
[7;23H#
[7;24H#
[7;25H#
[7;26H#
[7;27H#
[7;28H#
[7;29H#
[7;30H#
[8;11H#
[8;12H#
[8;13H#
[8;14H#
[8;15H#
[8;16H#
[8;17H#
[8;18H#
[8;19H#
[8;20H#
[8;21H#
[8;22H#
[8;23H#
[8;24H#
[8;25H#
[8;26H#
[8;27H#
[8;28H#
[8;29H#
[8;30H#
[9;11H#
[9;12H#
[9;13H#
[9;14H#
[9;15H#
[9;16H#
[9;17H#
[9;18H#
[9;19H#
[9;20H#
[9;21H#
[9;22H#
[9;23H#
[9;24H#
[9;25H#
[9;26H#
[9;27H#
[9;28H#
[9;29H#
[9;30H#
[10;11H#
[10;12H#
[10;13H#
[10;14H#
[10;15H#
[10;16H#
[10;17H#
[10;18H#
[10;19H#
[10;20H#
[10;21H#
[10;22H#
[10;23H#
[10;24H#
[10;25H#
[10;26H#
[10;27H#
[10;28H#
[10;29H#
[10;30H#
[11;11H#
[11;12H#
[11;13H#
[11;14H#
[11;15H#
[11;16H#
[11;17H#
[11;18H#
[11;19H#
[11;20H#
[11;21H#
[11;22H#
[11;23H#
[11;24H#
[11;25H#
[11;26H#
[11;27H#
[11;28H#
[11;29H#
[11;30H#
[12;11H#
[12;12H#
[12;13H#
[12;14H#
[12;15H#
[12;16H#
[12;17H#
[12;18H#
[12;19H#
[12;20H#
[12;21H#
[12;22H#
[12;23H#
[12;24H#
[12;25H#
[12;26H#
[12;27H#
[12;28H#
[12;29H#
[12;30H#
[13;11H#
[13;12H#
[13;13H#
[13;14H#
[13;15H#
[13;16H#
[13;17H#
[13;18H#
[13;19H#
[13;20H#
[13;21H#
[13;22H#
[13;23H#
[13;24H#
[13;25H#
[13;26H#
[13;27H#
[13;28H#
[13;29H#
[13;30H#
[14;11H#
[14;12H#
[14;13H#
[14;14H#
[14;15H#
[14;16H#
[14;17H#
[14;18H#
[14;19H#
[14;20H#
[14;21H#
[14;22H#
[14;23H#
[14;24H#
[14;25H#
[14;26H#
[14;27H#
[14;28H#
[14;29H#
[14;30H#
[15;11H#
[15;12H#
[15;13H#
[15;14H#
[15;15H#
[15;16H#
[15;17H#
[15;18H#
[15;19H#
[15;20H#
[15;21H#
[15;22H#
[15;23H#
[15;24H#
[15;25H#
[15;26H#
[15;27H#
[15;28H#
[15;29H#
[15;30H#
[101;1H